echopype/commongrid/api.py

```diff
@@ -163,6 +168,242 @@ def _parse_x_bin(x_bin: str, x_label="range_bin") -> float:
+
+    Parameters
+    ----------
+    ds_Sv : xr.Dataset
+        A Dataset containing ``Sv`` and other variables,
+        depending on computation performed.
+
+        For MVBS computation, this must contain ``Sv`` and ``echo_range`` data
+        with coordinates ``channel``, ``ping_time``, and ``range_sample``
+        at bare minimum.
+        Or this can contain ``Sv`` and ``depth`` data with similar coordinates.
+
+        For NASC computatioon this must contain ``Sv`` and ``depth`` data
+        with coordinates ``channel``, ``distance_nmi``, and ``range_sample``.
+    range_interval: pd.IntervalIndex or np.ndarray
+        1D array or interval index representing
+        the bins required for ``range_var``
+    x_interval : pd.IntervalIndex or np.ndarray
+        1D array or interval index representing
+        the bins required for ``ping_time`` or ``distance_nmi``.
+    x_var : {'ping_time', 'distance_nmi'}, default 'ping_time'
+        The variable to use for x binning. This will determine
+        if computation is for MVBS or NASC.
+    range_var: {'echo_range', 'depth'}, default 'echo_range'
+        The variable to use for range binning.
+        Either ``echo_range`` or ``depth``.
+
+        **For NASC, this must be ``depth``.**
+    method: str
+        The flox strategy for reduction of dask arrays only.
+        See flox `documentation <https://flox.readthedocs.io/en/latest/implementation.html>`_
+        for more details.
+    **flox_kwargs
+        Additional keyword arguments to be passed
+        to flox reduction function.
+
+    Returns
+    -------
+    xr.Dataset
+        The MVBS or NASC dataset of the input ``ds_Sv`` for all channels
+    """
+    # Check if x_var is valid, currently only support
+    # ping_time and distance_nmi, which indicates
+    # either a MVBS or NASC computation
+    if x_var not in ["ping_time", "distance_nmi"]:
+        raise ValueError("x_var must be 'ping_time' or 'distance_nmi'")
+
+    # Set correct range_var just in case
+    if x_var == "distance_nmi" and range_var != "depth":
+        logger.warning("x_var is 'distance_nmi', setting range_var to 'depth'")
+        range_var = "depth"
+
+    # Determine range_dim for NASC computation,
+    # this is not used for MVBS computation
+    range_dim = "range_sample"
+    if range_dim not in ds_Sv.dims:
+        range_dim = "depth"
+
+    # average should be done in linear domain
+    sv = ds_Sv["Sv"].pipe(_log2lin)
+
+    # Get positions if exists
+    # otherwise just use an empty dataset
+    ds_Pos = xr.Dataset(attrs={"has_positions": False})
+    if all(v in ds_Sv for v in POSITION_VARIABLES):
+        ds_Pos = xarray_reduce(
+            ds_Sv[POSITION_VARIABLES],
+            ds_Sv[x_var],
+            func="nanmean",
+            expected_groups=(x_interval),
+            isbin=True,
+            method=method,
+        )
+        ds_Pos.attrs["has_positions"] = True
```

Comment:
Let's move this block to a separate function called in `compute_MVBS` or `compute_NASC`, so that `get_x_along_channels` does not have a "hidden" handling of lat/lon positions.

Reply:
Ok! :D